docs/WebTransport.md

```diff
@@ -0,0 +1,246 @@
+# Using WebTransport in Kestrel
```

Comment:
This is a really good doc, it will make a great starting point for both a blog post and conceptual docs.

Reply:
Ok. I am happy you think so because seems like there is a ton of comments about it. I was unsure if maybe it wasn't a good idea to include in this PR or if it should be left as a follow up,